components/TokenApprovalList/index.tsx

```diff
@@ -0,0 +1,368 @@
+            entries.map(item => {
+              const txnHash = item.transaction_hash
+              const lastUpdateTime = formatDatetime(Date.parse(item.block?.timestamp), 'YYYY-MM-DD HH:mm:ss')
+              const isOwner = item.token_owner_address_hash === connectAccount?.account
+
+              return (
+                <tr key={txnHash}>
+                  <td>
+                    <div style={{ display: 'flex', alignItems: 'center' }}>
+                      <Address address={txnHash} />
+                    </div>
+                  </td>
+                  <td>
+                    {lastUpdateTime ? (
+                      <div className={styles.lastUpdateTime} title={lastUpdateTime}>
+                        {lastUpdateTime}
+                      </div>
+                    ) : (
+                      '-'
+                    )}
+                  </td>
+                  <td>
+                    <div className={styles.asset}>
+                      <NextLink href={`/token/${item.udt.id}`}>
+                        <a className={styles.logo}>
+                          <TokenLogo name={item.udt.name} logo={item.udt.icon} />
+                        </a>
+                      </NextLink>
+                      <NextLink href={`/token/${item.udt.id}`}>
+                        <a>{parseTokenName(item.udt.name).name}</a>
+                      </NextLink>
+                    </div>
+                  </td>
+                  <td>{item.udt.eth_type.replace('C', 'C-')}</td>
+                  <td>{item.type === GraphQLSchema.ApprovalType.Approval ? 'approve' : 'approve for all'}</td>
+                  <td>
+                    <Address address={item.spender_address_hash} />
+                  </td>
+                  <td>
+                    <Tooltip title={t('click_to_revoke')} placement="top">
+                      <button
+                        className={styles.revoke}
+                        disabled={!connector.ready || isRevokeTxnLoading}
+                        onClick={() => {
+                          const mapEthTypeToABI = {
+                            [GraphQLSchema.EthType.ERC20]: {
+                              address: item.token_contract_address_hash,
+                              abi: erc20ABI,
+                              function: 'approve',
+                              args: [item.spender_address_hash, 0],
+                            },
+                            [GraphQLSchema.EthType.ERC721]: {
+                              address: item.token_contract_address_hash,
+                              abi: erc721ABI,
+                              function: 'setApprovalForAll',
+                              args: [item.spender_address_hash, false],
+                            },
+                            [GraphQLSchema.EthType.ERC1155]: {
+                              address: item.token_contract_address_hash,
+                              abi: erc721ABI,
```

Comment:
`erc1155ABI`

Reply:
I found there is no existing `erc1155ABI` from ether.js (and not sure how to get one), so I used erc721ABI, since the function we need is `setApprovalForAll` and it's the same in both erc721 and erc1155.

(Maybe we can handwrite an ABI with only setApprovalForAll method?)

Reply:
> I found there is no existing `erc1155ABI` from ether.js (and not sure how to get one), so I used erc721ABI, since the function we need is `setApprovalForAll` and it's the same in both erc721 and erc1155.
> 
> (Maybe we can handwrite an ABI with only setApprovalForAll method?)

I recommend adding ABI of ERC 1155 to make it clear and extensible, and ABI of ERC 1155 can be exported by remix or hardhat easily.